proj-xcode/Classes/networking/private/client/PA2HttpClient.m

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright 2018 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "PA2HttpClient.h"
+#import "PA2AsyncOperation.h"
+#import "PA2PrivateMacros.h"
+
+@implementation PA2HttpClient
+{
+	dispatch_queue_t _completionQueue;
+	dispatch_queue_t _dispatchQueue;
+}
+
+- (instancetype) initWithConfiguration:(PA2ClientConfiguration*)configuration
+					   completionQueue:(dispatch_queue_t)completionQueue
+							   baseUrl:(NSString*)baseUrl
+								helper:(id<PA2PrivateCryptoHelper>)helper
+{
+	self = [super init];
+	if (self) {
+		_configuration = configuration;
+		_completionQueue = completionQueue;
+		_cryptoHelper = helper;
+
+		// Prepare baseUrl (without the last separator)
+		if ([baseUrl hasSuffix:@"/"]) {
+			_baseUrl = [baseUrl substringToIndex:baseUrl.length - 1];
+		} else {
+			_baseUrl = baseUrl;
+		}
+		
+		// Prepare queues.
+		// Note that dispatch queue is serial, but NSOPerationQueue is concurrent (by default)
+		_dispatchQueue = dispatch_queue_create("PA2HttpClient", DISPATCH_QUEUE_SERIAL);
+		_operationQueue = [[NSOperationQueue alloc] init];
```

Comment:
You can use `_operationQueue.maxConcurrentOperationCount = 1;` to create serial `NSOperationQueue`

Reply:
That's intended. I don't want to bring serialization queue now.